utils/managed_bytes.hh

```diff
@@ -356,6 +357,14 @@ public:
         }
     }
 
+    // Conforms to FragmentRange<managed_bytes> concept
+    using fragment_type = bytes_view;
+    using fragment_iterator = managed_bytes_view_fragment_iterator;
+    using const_iterator = fragment_iterator;
+    const_iterator begin() const noexcept;
+    const_iterator end() const noexcept;
```

Comment:
This is wrong, users of managed_bytes will expect that iterators iterate over bytes, not fragments, since managed_bytes is a bytes replacement.

Reply:
The fragment_type is `bytes_view`, isn't that expected when iterating over `bytes`?

Reply:
When iterating over bytes, you get a byte, not a byte_view.

Reply:
I like the adaptor idea, we can create a friend class that does the adaptation using the managed_bytes internals directly.

Reply:
Yes, and managed_bytes[_view]::as_fragment_range() can return it.

Reply:
I don't think we can have it as static member. The adaptor has to own the `managed_bytes{_view}` instance it operates on. Adaptors are expected to own the range they adapt. One of the bugs I fixed in IMR was caused by some owning the range some not.

Reply:
do we also need the byte iterator?

Reply:
> I don't think we can have it as static member. The adaptor has to own the `managed_bytes{_view}` instance it operates on. Adaptors are expected to own the range they adapt. One of the bugs I fixed in IMR was caused by some owning the range some not.

I implemented it as returning a `managed_bytes_fragment_range_view(*this)` that holds a copy of the `managed_bytes_view`.

Reply:
A copy works for the view, but we also need support for the `managed_bytes` itself.

Reply:
> do we also need the byte iterator?

I don't think so.

Reply:
Or maybe we do, I just came across some code that would be much better with it.